dev/Common/Logging.h

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation and Contributors.
+// Licensed under the MIT License.
+
+#if !defined(__LOGGING_H)
+#define __LOGGING_H
+
+#include <Windows.h>
+#include <source_location>
+#include <sstream>
+
+namespace Common::Logging
+{
+
+    // use std::format in msg parameter to have a printf like varargs syntax for the debug message
+    inline void DebugLog(const wchar_t* msg,  const std::source_location location = std::source_location::current())
+    {
+        
+        std::wstringstream filename ;
+        filename << location.file_name(); // converting char to wstring (and wchar_t implicitly)
```

Comment:
probably don't need the stream for this

Reply:
all other solutions I checked are error-prone and complicated. like `mbstowcs_s`. Any other way you can suggest ?

Reply:
can't you just pass const char* file_name() to std::format directly?https://en.cppreference.com/w/cpp/utility/source_location/file_name